Pipe commands into OutputFilter

For each project, we use a Makefile to manage our (VUnit) simulation and the Vivado project (create/open, build, export, etc.), among other things.

We already have in place some functionality to achieve some of the functionality provided by the OutputFilter. For example, we use pipe commands to pass the output of our console to the `sed` command, which then colorizes warnings, critical warnings, errors and failures. This is quite useful to identify dangerous warnings in our local console or CI/CD console.

We also use pipe commands to write the console output into log files, including filtered log files that contain only warnings using the `tee` command.

I see the OutputFilter as something that provides some other extra features and probably more flexibility for expansion. 

Do you think I could use this approach to pipe the output of our Makefile commands into the OutputFilter? Is this something already supported or planned to be supported?

Hi @Paebbels,

I gave it a try with version  `pyedaa.outputfilter==0.5.1`.

I got the following bug:

```bash
$ make save_project
Cannot create a file when that file already exists.
Junction created for C:\_build\etlw_001 <<===>> ..
[FATAL] An unknown or unhandled exception reached the topmost exception handler!
  Exception type:       Exception
  Exception message:    Unknown LineKind 'LineKind.ProcessorError|Empty' for line 1.
  Caused in:            GetColorOfLine(...) in file 'C:\Python\Python313\Lib\site-packages\pyEDAA\OutputFilter\CLI\Vivado.py' at line 336
  ------------------------------------------------------------------------------------------------------------------------
  File "C:\Python\Python313\Lib\site-packages\pyEDAA\OutputFilter\CLI\__init__.py", line 170, in main
    program.Run()
    ~~~~~~~~~~~^^
  File "C:\Python\Python313\Lib\site-packages\pyEDAA\OutputFilter\CLI\__init__.py", line 105, in Run
    ArgParseHelperMixin.Run(self)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "C:\Python\Python313\Lib\site-packages\pyTooling\Attributes\ArgParse\__init__.py", line 332, in Run
    self._ParseArguments()
    ~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Python\Python313\Lib\site-packages\pyTooling\Attributes\ArgParse\__init__.py", line 344, in _ParseArguments
    self._RouteToHandler(parsed)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "C:\Python\Python313\Lib\site-packages\pyTooling\Attributes\ArgParse\__init__.py", line 348, in _RouteToHandler
    args.func(self, args)
    ~~~~~~~~~^^^^^^^^^^^^
  File "C:\Python\Python313\Lib\site-packages\pyEDAA\OutputFilter\CLI\Vivado.py", line 106, in HandleVivado
    writeOutput(line)
    ~~~~~~~~~~~^^^^^^
  File "C:\Python\Python313\Lib\site-packages\pyEDAA\OutputFilter\CLI\Vivado.py", line 115, in _WriteColoredOutput
    color = self.GetColorOfLine(line)
  File "C:\Python\Python313\Lib\site-packages\pyEDAA\OutputFilter\CLI\Vivado.py", line 336, in GetColorOfLine
    raise Exception(f"Unknown LineKind '{line._kind}' for line {line._lineNumber}.")
  ------------------------------------------------------------------------------------------------------------------------
  Please report this bug at GitHub: https://github.com/edaa-org/pyEDAA.OutputFilter/issues
  ------------------------------------------------------------------------------------------------------------------------
========================================================================================================================
                                          pyEDAA.OutputFilter Service Program
========================================================================================================================
make: *** [save_project] Error 1
```

The OutputFilter was called as follows:

`output_filter=$(PY_VENV_DIR)/Scripts/pyedaa-outputfilter`
`$(vivado) <vivado_arguments> |& $(output_filter) vivado --stdin --colored`

--- a/pyEDAA/OutputFilter/__init__.py
+++ b/pyEDAA/OutputFilter/__init__.py
@@ -34,7 +34,7 @@ __author__ =    "Patrick Lehmann"
 __email__ =     "Paebbels@gmail.com"
 __copyright__ = "2014-2026, Patrick Lehmann"
 __license__ =   "Apache License, Version 2.0"
-__version__ =   "0.8.0"
+__version__ =   "0.9.0"
 __keywords__ =  ["cli", "abstraction layer", "eda", "filter", "classification"]
 
 from datetime              import datetime

--- a/run.ps1
+++ b/run.ps1
@@ -33,7 +33,7 @@ Param(
 )
 
 $PackageName = "pyEDAA.OutputFilter"
-$PackageVersion = "0.8.0"
+$PackageVersion = "0.9.0"
 
 # set default values
 $EnableDebug =        [bool]$PSCmdlet.MyInvocation.BoundParameters["Debug"]

